jax/_src/numpy/lax_numpy.py

```diff
@@ -1368,6 +1368,82 @@ def gradient(f, *varargs, axis: Optional[Union[int, Tuple[int, ...]]] = None,
 def isrealobj(x):
   return not iscomplexobj(x)
 
+@partial(jit, static_argnums=1)
+class _RankWarning(UserWarning):
+    """
+    Issued by `polyfit` when the Vandermonde matrix is rank deficient.
+    For more information, a way to suppress the warning, and an example of
+    `RankWarning` being issued, see `polyfit`.
+    """
+    pass
+
+@_wraps(np.polyfit)
+def polyfit(x, y, deg, rcond=None, full=False, w=None, cov=False):
+  """
+  Note: Unlike numpy, the function will not warn on rank reduction, which indicates an ill conditioned matrix
+  """
+  _check_arraylike("polyfit", x, y)
+  deg = core.concrete_or_error(int, deg, "deg must be int")
+  order = deg + 1
+  # check arguments.
+  if deg < 0:
+      raise ValueError("expected deg >= 0")
```

Comment:
Here and throughout, please use 2-space indentation for consistency.

Reply:
To make it consistent with other functions in the file, changing it to one space indentation